### ISSUE TITLE
How to fix 400 error ?

### ISSUE THREAD

**Issue by golubvladimir** [ivanmarban/winston-telegram#16]
Function for send messages to telegram.

```
const logger = async function (level, error) {
  try {
    if (['info', 'error'].indexOf(level) !== -1) {
      await sendToTelegram(
        level,
        error.name,
        error.message,
        error.fileName,
        error.lineNumber,
        error.columnNumber,
        error.stack,
      );
    }
  } catch (e) {
    console.log(e);
  }
};

const sendToTelegram = function send(level = '-', name = '-',
                                     info = '-', fileName = '-', lineNumber = 0,
                                     columnNumber = 0, stack = '-') {
  const dateMsg = new Date();
  const dateMsgFormat = `
          ${dateMsg.getHours()}:${dateMsg.getMinutes()}:${dateMsg.getSeconds()}
          ${dateMsg.getDay()}.${dateMsg.getMonth()}.${dateMsg.getFullYear()}
        `
        .replace(/\r?\n/g, "")
        .replace(/\s{2,}/g, ' ');

  if (incrementCounter()) {

    loggerWinston.log({
      level: level,
      message: 'level: ' + String(level) + '\n' +
              'data: ' + String(dateMsgFormat) + '\n' +
              'name: ' + String(name) + '\n' +
              'information: ' + String(info + ' ') + '\n' +
              'file name: ' + String(fileName) + '\n' +
              'line number: ' + Number(lineNumber) + '\n' +
              'column number: ' + Number(columnNumber) + '\n'
              'stack: ' + String(stack)
    });
  }
};
```

When i call her, 
```
logger('info', {name: 'Информационное сообщение', message: 'Информационное сообщение с BackEnd'});
```
I get an error.
```
Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('400: Bad Request: message text is empty')
```
But, when I comment line, it works.

```
    loggerWinston.log({
      level: level,
      message: 'level: ' + String(level) + '\n' +
              'data: ' + String(dateMsgFormat) + '\n' +
              'name: ' + String(name) + '\n' +
              //'information: ' + String(info + ' ') + '\n' + // commnent line
              'file name: ' + String(fileName) + '\n' +
              'line number: ' + Number(lineNumber) + '\n' +
              'column number: ' + Number(columnNumber) + '\n'
              'stack: ' + String(stack)
    });
```

**Comment by golubvladimir** [comment]
I found a mistake.

It works.

```
        this.logger({
          level: 'info',
          error: {
            name: 'Информационное сообщение',
            message: 'Проверка'
          }
        })
```

It doesn't work.

```
        this.logger({
          level: 'info',
          error: {
            name: 'Информационное сообщение',
            message: 'Проверка сообщения сообщения сообщения'
          }
        })
```

Telegram cannot process such a length of string.

**Comment by ivanmarban** [comment]
What is the length of the message? 

This message var:

``` javascript
loggerWinston.log({
      level: level,
     message: 'level: ' + String(level) + '\n' +
              'data: ' + String(dateMsgFormat) + '\n' +
              'name: ' + String(name) + '\n' +
              'information: ' + String(info + ' ') + '\n' +
              'file name: ' + String(fileName) + '\n' +
              'line number: ' + Number(lineNumber) + '\n' +
              'column number: ' + Number(columnNumber) + '\n' + 
              'stack: ' + String(stack)
    });
```

I think that Telegram has a maximum message length of 4096 characters.

**Comment by golubvladimir** [comment]
@ivanmarban  
1) 156
2) 186

**Comment by golubvladimir** [comment]
Add before 

```
  send (messageText) {
    const self = this

    console.log({msg: messageText.length});

    const requestData = JSON.stringify({
      chat_id: this.chatId,
      text: messageText,
      disable_notification: this.disableNotification
    })
```

Result
```
{ msg: 198 }
{ msg: 163 }
```

**Comment by ivanmarban** [comment]
What about requestData ? Could you stop debugging at that point to see the content of that var ?

**Comment by golubvladimir** [comment]
@ivanmarban 

```
requestData: '{"chat_id":"-...","text":"[info] level: info\\ndata:  10:33:30 2.11.2019 \\nname error: Информационное сообщение\\ndescription error: Проверка\\nfile name: -\\nline number: 0\\ncolumn number: 0\\nstack: -","disable_notification":false}'
}
{
requestData: '{"chat_id":"-...","text":"[info] level: info\\ndata:  10:33:30 2.11.2019 \\nname error: Информационное сообщение\\ndescription error: Проверка сообщение сообщения сообщения\\nfile name: -\\nline number: 0\\ncolumn number: 0\\nstack: -","disable_notification":false}'
}
```

**Comment by golubvladimir** [comment]
@ivanmarban try %20 for line break. It doesn't work. Error repeated

**Comment by golubvladimir** [comment]
I try this
```
https://api.telegram.org/bot.../sendMessage?chat_id=-...&text=[info] level: info%20data:  10:40:40 2.11.2019 %20name error: Информационное сообщение%20description error: Проверка сообщение сообщения сообщения%20file n
ame: -%20line number: 0%20column number: 0%20stack: -
```
or 
```
https://api.telegram.org/bot.../sendMessage?chat_id=-...&text=[info] level: info\ndata:  10:40:40 2.11.2019 \nname error: Информационное сообщение\ndescription error: Проверка 
```
or
```
https://api.telegram.org/bot.../sendMessage?chat_id=-...&text=[info] level: info\\ndata:  10:40:40 2.11.2019\\nname error: Информационное сообщение\\ndescription error: Проверка 
```
It works, but line break doesn't work.

**Comment by golubvladimir** [comment]
@ivanmarban 
This problem is related to the Russian language.

**Comment by ivanmarban** [comment]
The problem is related with ```'Content-Length': requestData.length``` header. It works sending the message without that header.

I need to figure out why.

**Comment by golubvladimir** [comment]
@ivanmarban 
Why is this heading? It is not in the documentation.

**Comment by ivanmarban** [comment]
https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

Applications should use this field to indicate the transfer-length of the message-body.

But the problem is not the header, it's about unicode characters of Cyrillic alphabet. The ```requestData.length``` is wrong while posting to Telegram API. Content length should be setted using ```Buffer.byteLength``` not ```string.length```.

**Comment by ivanmarban** [comment]
Fixed on 2.2.1
